### PR TITLE
feat: Add `allow_missing_columns` option to `read/scan_parquet`

### PR DESCRIPTION
**Pull request by nameexhaustion**
~~Let me know if there is a better name for this parameter~~

### CODE REVIEW

File: crates/polars-io/src/parquet/read/read_impl.rs [hunk 1]
```diff
@@ -384,20 +391,30 @@ fn rg_to_dfs_prefiltered(
                     .then(|| calc_prefilter_cost(&filter_mask))
                     .unwrap_or_default();
 
+                #[cfg(debug_assertions)]
```
Comment (coastalwhite) [inline crates/polars-io/src/parquet/read/read_impl.rs:394]: nit: `if cfg!(debug_assertions)`

**Comment by nameexhaustion** [comment]
I bumped the minor version for `NodeTraverser` as this exposes a new field in `FileScanOptions`

**Comment by coastalwhite** [comment]
I feel like `strict_schema` is a better name for this, but that is a bit subjective

**Comment by ritchie46** [comment]
> How about allow_missing_columns?

Yeah, let's go for that one. And default to `False`.

**Comment by coastalwhite** [comment]
I think this should also be added to read_parquet if possible

**Comment by ritchie46** [comment]
Nice, can we also refer to the solution in the error when the schema resolving fails in strict mode?

**Pull request merged**